### PR TITLE
Various ABLDuck improvements

### PR DESCRIPTION
**Pull request by clement-brodu**
# Description

Many improvement for class documentation.

You can see the result in my demo project that was generates with all my PR:  
* [Sources](https://github.com/clement-brodu/OEdoc-sample)
* [Doc](https://clement-brodu.github.io/OEdoc-sample/)

Here are all my changes :  
- fix: static elements not correct : [sample](https://clement-brodu.github.io/OEdoc-sample/#!/class/static.StaticCommentClass)
- feat: add static attribute for temp-table and dataset  
- feat: improve ABL comment parser : [sample](https://clement-brodu.github.io/OEdoc-sample/#!/class/property.PropertyComment)
- feat: display extent and format for tt fields : [sample](https://clement-brodu.github.io/OEdoc-sample/#!/procedure/proc/ProcSample.p-temptable-ttSample)
- feat: allow adding extra Tag in comment
- feat: improve enums documentation : [sample](https://clement-brodu.github.io/OEdoc-sample/#!/class/header.EnumHeader1)
- feat: display the get set modifiers if usefull : [sample](https://clement-brodu.github.io/OEdoc-sample/#!/class/property.PropertyComment-property-property10)

I had unit tests for all features and also for PR #392 and #389.  

For my demo project, I also customize the theme/template to change icons and [link to external class](https://clement-brodu.github.io/OEdoc-sample/#!/class/property.PropertyType).   
I will create another PR for this.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My branch is started from the latest commit in master
- [x] My commit history is clean
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by gquerret** [comment]
Not sure if you automatically get the notification from GitHub, but there are test failures.

**Comment by clement-brodu** [comment]
Yes, I saw it. 

It's a line endings problem in sources files that are used to test.

I will try to fix it.
Like somebody said "It works on my machine" :)

**Comment by clement-brodu** [comment]
I made a fix to convert crlf to lf in comments.

It should avoid platform and git issues.

File: src/java/eu/rssw/pct/oedoc/ClassDocumentationVisitor.java [hunk 1]
```diff
@@ -448,4 +469,12 @@ private static String getDataTypeName(TypeName typeName) {
             return typeName.getName();
 
     }
+
+    private String ltrim(String s) {
```
Comment (gquerret) [inline src/java/eu/rssw/pct/oedoc/ClassDocumentationVisitor.java:473]: The string `   Test Test2` will return `Test2`, not `Test Test2` I'd expect from an ltrim function. Is that expected ?

File: src/java/za/co/mip/ablduck/ABLDuck.java [hunk 2]
```diff
@@ -169,7 +169,6 @@ public void setDataFilesOnly(Boolean dataOnly) {
 
     @Override
     public void execute() {
-        checkDlcHome();
```
Comment (gquerret) [inline src/java/za/co/mip/ablduck/ABLDuck.java:173]: Why did you remove that ? Every PCT task should call this method around the beginning of execute()

Comment (clement-brodu) [reply src/java/za/co/mip/ablduck/ABLDuck.java:173]: We don't need DLC path to execute this task, so I just thougth we could remove it because in this case it's useless.
_(and it alowed me to test this task with a computer without openedge on it)_

But I can put it back if you prefer :)

File: src/java/eu/rssw/pct/oedoc/ClassDocumentationVisitor.java (hunk 1 above)
Comment (clement-brodu) [reply src/java/eu/rssw/pct/oedoc/ClassDocumentationVisitor.java:473]: I just copy paste it from `ClassDocumentationVisitor`.

I tried the code juste now and `Test Test2` in input will return `Test Test2`

File: src/java/za/co/mip/ablduck/ABLDuck.java (hunk 2 above)
Comment (gquerret) [reply src/java/za/co/mip/ablduck/ABLDuck.java:173]: You're right.

File: src/java/eu/rssw/pct/oedoc/ClassDocumentationVisitor.java (hunk 1 above)
Comment (gquerret) [reply src/java/eu/rssw/pct/oedoc/ClassDocumentationVisitor.java:473]: Right :)

**Comment by clement-brodu** [comment]
I don't understand, you merged my MR but I don't find my code in master branch.

Do you know why ?

**Comment by gquerret** [comment]
That's really interesting, I don't see it either...
I'll find the commits and push again.

**Comment by clement-brodu** [comment]
I rebase the branch If you want

**Comment by gquerret** [comment]
Pushed again. Please verify everything is there.

**Comment by clement-brodu** [comment]
It looks ok for this one, but now, it's the #398 that disappeared !I don't understand what happened.

I rebase my [branch pct-ablduck-template ](https://github.com/clement-brodu/pct/tree/feat/pct-ablduck-template) on the new master.

**Comment by gquerret** [comment]
@clement-brodu I had to do a force push between the pull requests because a few files from ABLDuck were originally pushed with wrong EOL. I think I probably did something wrong.  As you know exactly what has to be merged in master, can you open a new PR (against 65f45590) with the changes ?

**Comment by clement-brodu** [comment]
Done : #399
